### PR TITLE
Refactor changed

### PR DESCRIPTION
**Pull request by mickhansen**

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 - [BUG] Error messages thrown by the db in languages other than english do not crash the app anymore (mysql, mariadb and postgres only) [#3567](https://github.com/sequelize/sequelize/pull/3567)
 - [FEATURE] All querys can be logged individually by inserting `logging: fn` in the query option.
 - [FEATURE] Partial index support for Postgres with `index.where`
+- [REFACTOR] `.changed()` now works proactively by setting a flag on `set` instead of matching reactively. Not that objects and arrays will not be checked for equality on set and will always result in a change if they are `set`.
```
Comment (janmeier) [inline changelog.md]: Not that ...

File: lib/instance.js [hunk 2]
```diff
@@ -5,8 +5,10 @@ var Utils = require('./utils')
   , BelongsToMany = require('./associations/belongs-to-many')
   , InstanceValidator = require('./instance-validator')
   , QueryTypes = require('./query-types')
+  , Dottie = require('dottie')
   , Promise = require('./promise')
   , _ = require('lodash')
+  , primitives = ['string', 'number', 'boolean']
```
Comment (janmeier) [inline lib/instance.js]: What about date?

File: lib/instance.js [hunk 3]
```diff
@@ -805,29 +823,27 @@ module.exports = (function() {
       force: false
     }, options || {});
 
-    var self = this;
-
     // This semi awkward syntax where we can't return the chain directly but have to return the last .then() call is to allow sql proxying
```
Comment (janmeier) [inline lib/instance.js]: The comment can be removed, right?

Comment (mickhansen) [inline lib/instance.js]: yea

File: lib/instance.js (hunk 2 above)
Comment (mickhansen) [inline lib/instance.js]: Can't compare date with `===`.

Comment (janmeier) [inline lib/instance.js]: Perhaps https://lodash.com/docs#isEqual on primitives then ?

Comment (mickhansen) [inline lib/instance.js]: I don't think it matters for dates. Unlses you compare on `valueOf` and i don't really see a reason to add the complexity.

**Comment by janmeier** [comment]
LGTM

**Pull request merged**